src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/Threads.java

```diff
@@ -157,21 +157,22 @@ private static synchronized void initialize(TypeDataBase db) {
         virtualConstructor.addMapping("MonitorDeflationThread", MonitorDeflationThread.class);
         virtualConstructor.addMapping("NotificationThread", NotificationThread.class);
         virtualConstructor.addMapping("StringDedupThread", StringDedupThread.class);
+        virtualConstructor.addMapping("AttachListenerThread", AttachListenerThread.class);
     }
 
     public Threads() {
         _list = VMObjectFactory.newObject(ThreadsList.class, threadListField.getValue());
     }
 
     /** NOTE: this returns objects of type JavaThread, CompilerThread,
-      JvmtiAgentThread, NotificationThread, MonitorDeflationThread and ServiceThread.
-      The latter four are subclasses of the former. Most operations
+      JvmtiAgentThread, NotificationThread, AttachListenerThread,
+      MonitorDeflationThread and ServiceThread. Most operations
```

Comment:
StringDedupThread was previously missed from this comment. Can you fix this also?

I think the "The latter four are subclasses of the former" comment should stay since all the text the follows relates to this. I just needs to be correct to say "seven" instead of "four".